### ISSUE TITLE
Warning "The following plugins do not have their version specified" displayed while it should not

### ISSUE THREAD

**Issue by glhez** [mojohaus/versions#358]
This may also relate to #148.

I have the following warnings, despite having the version correctly set:

```
[WARNING] The following plugins do not have their version specified:
[WARNING]   com.github.glhez:eclipse-settings-maven-plugin ........... 3.1.0
...
[WARNING] The following plugins do not have their version specified:
[WARNING]   maven-compiler-plugin .................................... 3.8.1
```

They both have different cause:.

**For the warning about com.github.glhez:eclipse-settings-maven-plugin**

The plugin is specified with a `groupId` being: the property `plugin.eclipse-settings.groupId` reference plugin which I have forked, and for which I changed to the `groupId` to avoid name clash.

```
          <plugin>
            <groupId>${plugin.eclipse-settings.groupId}</groupId>
            <artifactId>eclipse-settings-maven-plugin</artifactId>
            <version>${plugin.eclipse-settings.version}</version>
          </plugin>
```

The problem is related to the way the maven-version-plugin is getting the plugin and their version:

The message is produced here: https://github.com/mojohaus/versions-maven-plugin/blob/1afa5c3c0b5783a7fec30d204832712915317645/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java#L591

A plugin is added in `pluginLockdowns` for this condition:

```
if ( version == null || !pluginsWithVersionsSpecified.contains( coords ) ) {
  ...
  pluginLockdowns.add( pad( compactKey( groupId, artifactId ), ... );
}
```

In a debug session:

- `version` is correctly set to 3.1.0, so it is not null.
- `pluginsWithVersionsSpecified` does not contains coords
  - The variable is set to `com.github.glhez:eclipse-settings-maven-plugin`
  - The set contains the following item: `${plugin.eclipse-settings.groupId}:eclipse-settings-maven-plugin`

So, the problem is "simple": the set was constructed without properties being resolved to their final value.

The set is built using `DisplayPluginUpdatesMojo::findPluginsWithVersionsSpecified(StringBuilder)` which takes the source pom as data, and use a `ModifiedPomXMLEventReader` to read the pom as XML content.

That the source of the error: since it read the pom as XML, without the API Maven provide (and which I don't know), the `groupId` is never resolved: it is left as I entered it in my pom, producing this warning.

The (temporary) solution to hide this warning is to use a static `groupId`.

**For the warning about maven-compiler-plugin**

My pom tree is the following:

- `parent`: define `pluginManagement` and set version to `maven-compiler-plugin`.
  - `java-parent`: define common stuff for `maven-compiler-plugin`.
    - `java-11`: define Java 11 specific stuff for `maven-compiler-plugin`.

The code is running in the same `if` than before: the version is correctly set to 3.8.1 but the `maven-compiler-plugin` is absent from the `pluginsWithVersionsSpecified`: the version is resolved from the first parent only and it does not take into account a top parent.

I suppose the Maven API does take that into account, but I recon there is a corner case with `pluginMaganement` and `plugin` dependencies.

The (temporary) solution to hide this warning is to define a property and force the version in each module that is used as a parent (because their child module may override a plugin definition).

**Comment by aoindustries** [comment]
Thanks for reopening this issue.  We have procedural workarounds to it, but it would be nice to have it eventually resolved.

**Comment by stefanseifert** [comment]
i assume this is a duplicate of #550, #562 is my proposal to fix the issue
see also https://github.com/mojohaus/versions-maven-plugin/issues/550#issuecomment-1043068835 about my rationale for the fix

**Comment by ajarmoniuk** [comment]
it might have been fixed by #886 -- need to check

**Comment by glhez** [comment]
I closed the issue because 1) I no longer have the problem 2) do not have a sample project to reproduce.
